app/services/oa_workflow_service.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class OaWorkflowService
+  def workflow
+    Publication.needs_doi_verification.each do |pub|
```

Comment:
Here's where I think we're going to need to set the publication's `oa_workflow_status` to be something like "automatic DOI verification pending" and then exclude publications with that status from the `needs_doi_verification` scope on `Publication`. Otherwise, depending on how often the cron job runs and how long it takes for `DoiVerificationJob` to attempt to verify a DOI, we could end up queuing up (and running) more instances of the same job for the same publication before the first one has a chance to finish.

I think we may as well go ahead and do that now.

Reply:
I think it may be possible to lose a publication if an error occurs after the `oa_workflow_status` is set and the doi_check ends up staying nil.  This may be something that almost never happens, since we already handle error in the client class, but I think it could still happen?  Or perhaps we want the error to stop the publication from proceeding/checking again, so we can address the error.  I'm actually not too sure about this one.

Reply:
Current plan is to still set the `oa_workflow_status` at the beginning but add a rescue block that will set `doi_verified` to false. This will kick it over for manual verification and keep it from getting queued for the job again.